core/src/main/java/io/neow3j/transaction/Transaction.java

```diff
@@ -84,8 +84,18 @@ public long getValidUntilBlock() {
         return validUntilBlock;
     }
 
+    public List<Signer> getSigners() {
+        return signers;
+    }
+
     public ScriptHash getSender() {
-        return sender;
+        if (signers.isEmpty()) {
+            throw new NoSuchElementException("This transaction does not contain any signer.");
+        }
+        return signers.stream()
+                .filter(signer -> signer.getScopes().contains(WitnessScope.FEE_ONLY))
+                .findFirst()
+                .orElse(signers.get(0)).getScriptHash();
```

Comment:
Can you please add a comment here?

Also: `.getScriptHash()` should be in the next line.